File: apps/mobile/lib/features/community/domain/models/sponsorship_proof_document.dart
import 'package:meta/meta.dart';

@immutable
class SponsorshipProofDocument {
  final String id;
  final String fileName;
  final String mimeType;
  final int fileSizeBytes;
  final String? presignedUrl;
  final DateTime uploadedAt;

  const SponsorshipProofDocument({
    required this.id,
    required this.fileName,
    required this.mimeType,
    required this.fileSizeBytes,
    this.presignedUrl,
    required this.uploadedAt,
  });
}

File: apps/api/HappyPaws.Domain/Entities/SponsorshipProofDocument.cs
using System;

namespace HappyPaws.Domain.Entities;

public class SponsorshipProofDocument
{
    public Guid Id { get; set; }
    public Guid PostId { get; set; }
    public string StorageKey { get; set; } = null!;
    public string FileName { get; set; } = null!;
    public string MimeType { get; set; } = null!;
    public int FileSizeBytes { get; set; }
    public DateTimeOffset UploadedAt { get; set; }

    public Post Post { get; set; } = null!;
}
